python_scripts/libaerie/products/product_parser.py

```diff
@@ -4,44 +4,103 @@
+            try:
+                self._fh = open(filename, "r")
+            except Exception as e:
+                logger.exception(e)
+        else:
+            logging.error("File not found: '%s'", filename)
+            raise FileNotFoundError("File not found: %s" % filename)
+
+        try:
+            self.filename = self._fh.name
+        except AttributeError:
+            self.filename = "Buffered_IO"
+        self.header_hash = None
 
     @abstractmethod
     def parse(self):
         pass
 
 
 class DsnViewPeriodPredLegacyDecoder(Decoder):
+    """
+    Manages state for decoding a DSN View Period Legacy report file
+
+    :ivar _fh: Private file handler used for reading data from the report
+    :vartype _fh: file object
+    :ivar header_hash: key / value store of the data contained in the header
+    :vartype header_hash: dict
+    :cvar HEADER_TIME_FORMAT: Format of the datetimes in the header
+    :vartype HEADER_TIME_FORMAT: str
+    :cvar EVENT_TIME_FORMAT: Format of the datetimes in the events
+    :vartype EVENT_TIME_FORMAT: str
+    """
 
     HEADER_TIME_FORMAT = "%Y-%jT%H:%M:%S"
     EVENT_TIME_FORMAT = "%y %j/%H:%M:%S"
 
     EVENT_RECORD_REGEX = "(?P<TIME>.{15}).(?P<EVENT>.{16}).(?P<SPACECRAFT_IDENTIFIER>.{3}).(?P<STATION_IDENTIFIER>.{2}).(?P<PASS>.{4}).(?P<AZIMUTH>.{5}).(?P<ELEVATION>.{5}).(?P<AZ_LHA_X>.{5}).(?P<EL_DEC_Y>.{5}).(?P<RTLT>.{10})"
 
-    def __init__(self, filename: str):
-        logger = logging.getLogger(__name__)
-
-        if not os.path.isfile(filename):
-            logging.error("DSN Viewperiod file not found: '%s'", filename)
-            raise FileNotFoundError("DSN Viewperiod file not found: %s" % filename)
+    def __init__(self, filename: Union[str | io.IOBase]):
+        """
+        Initialize a DsnViewPeriodPredLegacyDecoder which reads information from DSN View Period files.
 
-        logger.info("Opening DSN Viewperiod file for Decoding: %s", filename)
-        try:
-            self._fh = open(filename, "r")
-        except Exception as e:
-            logger.exception(e)
+        :param filename: Filepath to View Period file.
+        :type filename: str | io.IOBase
+        """
 
-        self.header_hash = None
+        logger = logging.getLogger(__name__)
+        logger.info("Opening DSN View Period file for Decoding: %s", filename)
+        super(DsnViewPeriodPredLegacyDecoder, self).__init__(filename)
 
     @classmethod
-    def chop_header_line(cls, line: str):
+    def chop_header_line(cls, line: str) -> tuple:
+        """
+        Converts a line from the View Period header into a key / value format
+
+        :param line: Line from the View Period file header
+        :type line: str
+        :return: key, value tuple of the header field and its value
+        :rtype: tuple
+        """
+
         logger = logging.getLogger(__name__)
 
+        # Left of "=" is the key, right is the value
         seg, value = line.split(" = ")
         value = value[:-1]
```

Comment:
Not sure I understand what's happening in this line - it may be worth a comment. Is value always one character?

Reply:
This splits the information in the header of the file into key / value components.  Each line of the header follows the format KEY = VALUE, e.g. MISSION_NAME = HUBBLE;.  value should contain "HUBBLE" in this case.  I'll add more comments to explain.

Reply:
I see - it's chopping off the trailing `;` :+1: The explanation you just gave would be great in a comment - I found the example particularly helpful